server/src/main/java/org/elasticsearch/inference/InferenceServiceConfiguration.java

```diff
@@ -50,39 +51,48 @@ public class InferenceServiceConfiguration implements Writeable, ToXContentObjec
      */
     private InferenceServiceConfiguration(
         String provider,
+        String name,
+        String icon,
         List<TaskSettingsConfiguration> taskTypes,
         Map<String, SettingsConfiguration> configuration
     ) {
         this.provider = provider;
+        this.name = name;
+        this.icon = icon;
         this.taskTypes = taskTypes;
         this.configuration = configuration;
     }
 
     public InferenceServiceConfiguration(StreamInput in) throws IOException {
         this.provider = in.readString();
+        this.name = in.readString();
+        this.icon = in.readString();
```

Comment:
We may need to use [TransportVersions](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/TransportVersions.java) to gate this call so that it returns this data if the client is running a supported version. Example [write](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/GetInferenceModelAction.java#L91) and [read](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/GetInferenceModelAction.java#L65).  Then these have to be treated as nullable =/

Reply:
I was debating whether to do that but thought that (as you said) since it was only called externally maybe we didn't need to but I'm happy to add it just to be safe if you think that's best!

Reply:
I added the transport version check in this commit: https://github.com/elastic/elasticsearch/pull/116738/commits/4b8b7af69f1a15d990bd245f8a54b7cdb3141e14